parser_test.go

```diff
@@ -0,0 +1,89 @@
+package terminal
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestParseSimpleXY(t *testing.T) {
+	s := parsedScreen("hello")
+	if err := assertTextXY(t, s, "hello", 5, 0); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestParseAfterCursorMovement(t *testing.T) {
+	s := parsedScreen("hello\x1b[4D!")
```

Comment:
i think that this probably comes with the territory, but relying on mushy human brains to parse these escape sequences to ensure that these tests are correct is pretty difficult... is there a way that we could replace this with something like 
```suggestion
	s := parsedScreen(fmt.Sprintf("hello%s!", moveCursorX(-4))) 
```
... or something? i guess the problem is that there isn't a clear difference between the ANSI escape (`\1b[4D`, i think?), and the literal text (`hello!` or `h!llo`, depending on how you look at it)

Reply:
Yeah that'd be nice, it is definitely hard to mentally parse.
I like your idea of using Sprintf for that.
(But I'm not going to do it in this PR 😜)